AutoCollection/ServerRequestParser.ts

```diff
@@ -169,6 +169,29 @@ class ServerRequestParser extends RequestParser {
         return value;
     }
 
+    private parseHeaders(request:http.ServerRequest, parentId?: string) {
+        this.rawHeaders = request.headers || (<any>request).rawHeaders;
+        this.userAgent = request.headers && request.headers["user-agent"];
+        this.sourceCorrelationId = Util.getCorrelationContextTarget(request, RequestResponseHeaders.requestContextSourceKey);
+
+        if (request.headers) {
+            this.correlationContext = request.headers[RequestResponseHeaders.correlationContextHeader];
+
+            if (request.headers[RequestResponseHeaders.requestIdHeader]) {
+                this.parentId = request.headers[RequestResponseHeaders.requestIdHeader];
+                this.requestId = CorrelationIdManager.generateRequestId(this.parentId);
+                this.correlationContext = request.headers[RequestResponseHeaders.correlationContextHeader];
+            } else {
+                // Legacy fallback
+                const rootId = request.headers[RequestResponseHeaders.rootIdHeader];
+                this.parentId = request.headers[RequestResponseHeaders.parentIdHeader];
+                this.requestId = CorrelationIdManager.generateRequestId(rootId || this.parentId || parentId);
+                this.correlationContext = null;
+            }
+            this.operationId = CorrelationIdManager.getRootId(this.requestId); 
```

Comment:
If no headers are present, then `this.parentId` will be null, but `this.requestId` will be `CorrelationIdManager.generateRequestId(parentId)` which will generate a hierarchical ID from the passed-in identifier (if any) or from nothing, so `this.requestId` should always be set.